tests/scripts/thread-cert/Cert_6_5_02_ChildResetReattach.py

```diff
@@ -28,59 +28,238 @@
-        self.nodes[ED].remove_allowlist(self.nodes[LEADER].get_addr64())
-
-        self.nodes[ED].reset()
-        self._setUpEd()
+        self.nodes[MTD].start()
         self.simulator.go(5)
-        self.nodes[ED].start()
+        self.assertEqual(self.nodes[MTD].get_state(), 'child')
+
+        self.nodes[LEADER].clear_allowlist()
+        self.nodes[MTD].clear_allowlist()
+        self.nodes[ROUTER].stop()
 
+        self.nodes[MTD].reset()
+        self._setUpMTD()
         self.simulator.go(5)
-        self.nodes[LEADER].add_allowlist(self.nodes[ED].get_addr64())
+        self.nodes[LEADER].add_allowlist(self.nodes[MTD].get_addr64())
         self.simulator.go(5)
-        self.assertEqual(self.nodes[ED].get_state(), 'child')
+        self.nodes[MTD].start()
+        self.simulator.go(10)
+        self.assertEqual(self.nodes[MTD].get_state(), 'child')
+        self.simulator.go(360)
+
+        self.collect_ipaddrs()
+        dut_addr = self.nodes[MTD].get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL)
+        self.nodes[LEADER].ping(dut_addr)
+        self.simulator.go(15)
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        LEADER_LLA = pv.vars['LEADER_LLA']
+        ROUTER = pv.vars['ROUTER']
+        DUT = pv.vars['DUT']
+        DUT_LLA = pv.vars['DUT_LLA']
+
+        # Step 1: Ensure topology is formed correctly
+
+        pv.verify_attached('ROUTER', 'LEADER')
+        pv.verify_attached('DUT', 'ROUTER', 'MTD')
+
+        # Step 4: DUT sends an MLE Child Update Request.
+        #         The following TLVs MUST be included in the Child Update
+        #         Request:
+        #             - Mode TLV
+        #             - Address Registration TLV (optional)
+        #         If the DUT is a SED, it MUST resume polling after sending
+        #         MLE Child Update.
+        _pkt = pkts.filter_wpan_src64(DUT).\
+            filter_wpan_dst64(ROUTER).\
+            filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+            filter(lambda p: {
+                              MODE_TLV
+                             } < set(p.mle.tlv.type)
+                   ).\
+            must_next()
+        with pkts.save_index():
+            if self.TOPOLOGY[MTD]['mode'] == '-':
+                # filter another 3 MLE_CHILD_UPDATE_REQUEST with the upper one
+                # (total 4) sent out continuously in one poll
+                for i in (1, 4):
+                    pkts.filter_wpan_src64(DUT).\
+                        filter_wpan_dst64(ROUTER).\
+                        filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                        must_next()
+                pkts.filter_wpan_src64(DUT).\
+                    filter_wpan_dst64(ROUTER).\
+                    filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                    filter(lambda p: p.sniff_timestamp - _pkt.sniff_timestamp <= POLL_PERIOD).\
```

Comment:
```suggestion
                    filter(lambda p: p.sniff_timestamp - _pkt.sniff_timestamp <= POLL_PERIOD + 0.1).\
```
nit: we can add some allowance if necessary.